pkg/apis/triggers/v1alpha1/trigger_types_convert_test.go

```diff
@@ -0,0 +1,108 @@
+package v1alpha1
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	v1 "k8s.io/api/core/v1"
+)
+
+func TestToEventListenerTrigger(t *testing.T) {
+	tt := []struct {
+		name string
+		in   TriggerSpec
+		out  EventListenerTrigger
+	}{
+		{
+			name: "Convert Empty Object",
+			in:   TriggerSpec{},
+			out:  EventListenerTrigger{},
+		},
+		{
+			name: "Convert Partial Object",
+			in: TriggerSpec{
+				Name: "foo",
+				Template: TriggerSpecTemplate{
+					Name: "baz",
+				},
+			},
+			out: EventListenerTrigger{
+				Name: "foo",
+				Template: EventListenerTemplate{
+					Name: "baz",
+				},
+			},
+		},
+		{
+			name: "Convert Full Object",
+			in: TriggerSpec{
+				Name: "a",
+				ServiceAccount: &v1.ObjectReference{
+					Name: "a",
+				},
+				Interceptors: []*TriggerInterceptor{{
+					Webhook: &WebhookInterceptor{},
+				}},
+				Bindings: []*TriggerSpecBinding{{
+					Name:       "a",
+					APIVersion: "b",
+					Kind:       "c",
+					Ref:        "d",
+					Spec: &TriggerBindingSpec{
+						Params: []Param{{
+							Name:  "a",
+							Value: "b",
+						}},
+					},
+				}},
+				Template: TriggerSpecTemplate{
+					Name:       "a",
+					APIVersion: "b",
+				},
+			},
+			out: EventListenerTrigger{
+				Name: "a",
+				ServiceAccount: &v1.ObjectReference{
+					Name: "a",
+				},
+				Interceptors: []*TriggerInterceptor{{
+					Webhook: &WebhookInterceptor{},
+				}},
+				Bindings: []*EventListenerBinding{{
+					Name:       "a",
+					APIVersion: "b",
+					Kind:       "c",
+					Ref:        "d",
+					Spec: &TriggerBindingSpec{
+						Params: []Param{{
+							Name:  "a",
+							Value: "b",
+						}},
+					},
+				}},
+				Template: EventListenerTemplate{
+					Name:       "a",
+					APIVersion: "b",
+				},
+			},
+		},
+		{
+			name: "Convert Empty Object",
+			in:   TriggerSpec{},
+			out:  EventListenerTrigger{},
+		},
```

Comment:
Can we remove this entire block because its already handled  [here](https://github.com/tektoncd/triggers/pull/676/files#diff-5e09b1bd1d763e83f4d558bb51254d59R16-R20)

Reply:
Whoops. done